_includes/markdown/JavaScript.md

```diff
@@ -155,6 +155,80 @@ On all new projects you should be using up to date JavaScript methodologies comb
 
 Some older projects that have not yet been upgraded may not have the capability to use the most modern techniques, but it is still important to have processes in place that allow us to grow the technology stack as a project matures. In these cases, you should still follow best practice recommendations even if the newest patterns are not yet available to you.
 
+### Secure Your Code
+
+In JavaScript, we often have to insert new elements with dynamic attributes and content into the DOM. A common way to do this is to use the [```innerHTML```](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) method like so:
+
+```javascript
+const someElement = document.getElementById( 'someElement' );
+const someUrl = 'https://someurl.com/';
+const someContent = 'Some content';
+
+someElement.innerHTML = `<div class="container"><a href="${ someUrl }">${ someContent }</a></div>`;
+```
+
+However, passing HTML strings to ```innerHTML``` and methods like it (including jQuery's commonly used ```append``` and ```html``` methods) can expose your code to cross-site scripting, also known as XSS—the most common security vulnerability in JavaScript. Because these methods evaluate strings passed to them as HTML, they can execute potentially harmful code. For example, what happens if ```someContent``` in the above example is ```<img src="fakeImage" onerror="alert( 'hacked' )" />```? The JavaScript in the ```onerror``` attribute is executed.
+
+There are several measures you can take to circumvent this XSS vulnerability:
+
+#### Use ```textContent``` instead of ```innerHTML```
+
+When setting the human-readable content of a single element, using [```textContent```](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) is safer than using ```innerHTML``` because it does not parse strings as HTML—meaning any malicious code passed to it will not be executed.
+
+```javascript
+const anotherElement = document.getElementById( 'anotherElement' );
+
+anotherElement.textContent = 'This will get displayed as human-readable text, even if it contains HTML-like <syntax>.';
+```
+
+#### Use the ```Element``` API instead of HTML-parsing methods
+
+When you need to create multiple DOM elements, use JavaScript's [```Element```](https://developer.mozilla.org/en-US/docs/Web/API/Element) API to create elements, set attributes and append them to the document. Creating your own elements and attributes will ensure that only those you explicitly define will make their way into the DOM. Refactoring our first example:
+
+```javascript
```

Comment:
@dkoo there's a ton of stuff in the Element API, rather than addressing a very small portion of it, let's just link off to MDN.